frontend/datetime.lua

```diff
@@ -0,0 +1,296 @@
+        if withoutSeconds then
+            time_string = time_string .. ":"
+        end
+        time_string = time_string:gsub(":", C_("Time", "h"), 1)
+        time_string = time_string:gsub(":", C_("Time", "m"), 1)
+        time_string = time_string:gsub("^00" .. C_("Time", "h"), "") -- delete leading "00h"
+        time_string = time_string:gsub("^00" .. C_("Time", "m"), "") -- delete leading "00m"
+        if time_string:find("^0%d") then
+            time_string = time_string:gsub("^0", "") -- delete leading "0"
+        end
+        if withoutSeconds and time_string == "" then
+            time_string = "0" .. C_("Time", "m")
+        end
+
+        if hmsFormat then
+            return withoutSeconds and time_string or (time_string .. C_("Time", "s"))
+        else
+            time_string = time_string:gsub(C_("Time", "m"), "'") -- replace m with '
+            return withoutSeconds and time_string or (time_string .. SECONDS_SYMBOL)
+        end
+    end
+end
+
+--- Converts seconds to a clock type (classic or modern), based on the given format preference
+--- "Classic" format calls secondsToClock, and "Modern" format calls secondsToHClock
+---- @string Either "modern" for 1h30'10" or "classic" for 1:30:10
+---- @bool withoutSeconds if true 1h30' or 1h30m, if false 1h30'10" or 1h30m10s
+---- @bool hmsFormat, modern format only, if true format 1h30m or 1h30m10s
+---- @bool withDays, if hours>=24 include days in clock string 1d12h10m10s
+---- @bool compact, if set removes all leading zeros (incl. units if necessary)
+---- @treturn string clock string in the specific format of 1h30', 1h30'10" resp. 1h30m, 1h30m10s
+function datetime.secondsToClockDuration(format, seconds, withoutSeconds, hmsFormat, withDays, compact)
+    if format == "modern" then
+        return datetime.secondsToHClock(seconds, withoutSeconds, hmsFormat, withDays, compact)
+    else
+         -- Assume "classic" to give safe default
+        return datetime.secondsToClock(seconds, withoutSeconds, withDays)
+    end
+end
+
+if jit.os == "Windows" then
+    --- Converts timestamp to an hour string
+    ---- @int seconds number of seconds
+    ---- @bool twelve_hour_clock
+    ---- @treturn string hour string
+    ---- @note: The MS CRT doesn't support either %l & %k, or the - format modifier (as they're not technically C99 or POSIX).
+    ----        They are otherwise supported on Linux, BSD & Bionic, so, just special-case Windows...
+    ----        We *could* arguably feed the os.date output to gsub("^0(%d)(.*)$", "%1%2"), but, while unlikely,
+    ----        it's conceivable that a translator would put something other that the hour at the front of the string ;).
+    function datetime.secondsToHour(seconds, twelve_hour_clock)
+        if twelve_hour_clock then
+            if os.date("%p", seconds) == "AM" then
+                -- @translators This is the time in the morning using a 12-hour clock (%I is the hour, %M the minute).
```

Comment:
Difficult, it may well be that every language other than English can only do `in the morning`/`in the afternoon` or `before noon/after noon` in full. Is this supposed to be short or is that fine?
```suggestion
                -- @translators Time in the morning using a 12-hour clock (%I hour, %M minute).
```